docs/development/design/new-sphinx-guides.rst

```diff
@@ -0,0 +1,186 @@
+
+   4. Appendix: Using version control
+
+      -  Install git (we will use it during the tutorial)
+      -  Write a barebones ``README.md`` (we will expand it later), a
+         ``.gitignore`` file (copied from gitignore.io), and a proper
+         ``LICENSE`` file
+
+         -  ``LICENSE.md``? ``LICENSE``? ``COPYING``? Whatever
+
+      -  Create the first commit for the project (yay!)
+
+2. First steps to document our project using Sphinx
+
+   1. Installing Sphinx
+
+      -  Activate our virtual environment
+      -  Install Sphinx inside the virtual environment
+      -  Check that ``sphinx-build --help`` works (yay!)
+
+   2. Creating the project layout
+
+      -  Run the sphinx-quickstart wizard: separate source and build
+         (``y``), project name, author name, and go
+      -  Check that the correct files are created (yay!)
+
+   3. Converting our documentation to local HTML
+
+      -  Build the HTML output using
+         ``sphinx-build -b -W html doc doc/_build/html``  [1]_
+      -  Navigate to ``doc/_build/html`` and launch an HTTP server
+         (``python -m http.server``)
+      -  Open http://localhost:8000 in a web browser, and see the HTML
+         documentation (yay!)
+
+   4. Making a change and regenerating the documentation
+
+      -  Add a sentence to ``index.rst`` (no need to explain in detail
+         what reST is at this stage)
+      -  Rebuild HTML ``sphinx-build -W -b html ...``
+      -  Observe that the HTML docs have changed (yay!) (we will have to
+         do this every time we change something)
+
+   5. Simplify documentation building by using Make
+
+      -  Install Make (and its Windows counterpart)
+      -  Make another trivial change
+      -  Build HTML doing ``cd doc && make html``
+      -  Observe that the HTML docs have changed (yay!)
+
+   6. Converting our documentation to PDF
+
+      -  Install TeX Live to create PDF output (should work on every
+         operative system)
+      -  Build LaTeX using ``make latexpdf``
+      -  See that the PDF appeared (yay!)
+
+   7. Appendix: PDF without LaTeX using rinoh (beta)
+
+3. Customizing Sphinx configuration
+
+   1. Changing the HTML theme
+
+      -  Install https://pypi.org/project/furo/
+      -  Change the ``html_theme`` in ``conf.py``
+      -  Rebuild the HTML documentation and observe that the theme has
+         changed (yay!)
+
+   2. Changing the PDF appearance
+
+      -  Add a ``latex_theme`` and set it to ``howto``
+      -  Rebuild ``make latexpdf``
+      -  Check that the appearance changed (yay!)
+
+   3. Enable an extension
+
+      -  Add a string to the ``extensions`` list in ``conf.py`` for
+         ``sphinx.ext.duration``
+      -  Rebuild the HTML docs ``make html`` and notice that now the
+         times are printed (yay!)
+
+4. Writing narrative documentation with Sphinx
+
+   -  First focus on ``index.rst``, gently introducing the learner to
+      reST and mentioning Semantic Line Breaks.
```

Comment:
I think we probably just want to introduce markdown here, if that's the plan, and never do much of an introduction to rst.

Reply:
Yeah, I am a bit conflicted. From the notes in the document:

> The fact that the ``index.rst`` created by ``sphinx-quickstart`` is written in reStructuredText gets in the way of teaching Markdown (MyST). On the other hand, users need to know reST anyway to write Python docstrings, so it is necessary to teach both.

In other words: I feel bad ignoring the `index.rst` that is created by the quickstart, and on the other hand reST is needed for docstrings anyway. Sure, we don't need to do a full reST introduction (see comments above), but even though I'd love to focus only on Markdown/MyST, perhaps it's not realistic for Python projects.

...Unless we don't have a Python library in mind, and this is geared towards narrative documentation only. But I think that would be too limiting.

Reply:
Switched to Markdown (by avoiding the `index.rst` created by `sphinx-quickstart`)